SetupDataPkg/Tools/GenNCCfgData.py

```diff
@@ -0,0 +1,323 @@
+            ord_dict['inst'] = data
+            ord_dict['order'] = idx
+            ord_dict['name'] = name
+            ord_dict['cname'] = name
+            ord_dict['value'] = data.format.object_to_string(data.value)
+            ord_dict['path'] = '.'.join([data.knob.namespace, name])
+            ord_dict['help'] = data.help
+            ret_list.append(ord_dict)
+
+        return ret_list
+
+    def sync_shim_and_schema (self):
+        for shim in self.knob_shim:
+            data = shim['inst']
+            shim['value'] = data.format.object_to_string(data.value)
+
+    def load_from_svd (self, path):
+        def handler(id, value):
+            if id is not None and id.startswith("Device.RuntimeData.RuntimeData"):
+                # this is the full svd, it is a base64 encoded bin
+                base64_val = value.strip()
+                bin_data = base64.b64decode(base64_val)
+                self.load_default_from_bin(bin_data, True)
+
+        a = DFCI_SupportLib()
+
+        a.iterate_each_setting(path, handler)
+
+    def load_default_from_bin (self, bin_data, variable_list_format):
+        var_list = read_vlist_from_buffer (bin_data)
+        # YAML variable is included in list, XML tree should not contain it
+        for var in var_list:
+          if str(var.guid).lower() == DFCI_SETTINGS_REQUEST_GUID.lower():
+            # delete from variable list
+            var_list.remove(var)
+        uefi_variables_to_knobs (self.schema, var_list)
+        self.sync_shim_and_schema ()
+
+    def generate_binary_array (self):
+        return vlist_to_binary (self.schema)
+
+    def generate_binary (self, bin_file_name):
+        bin_file = open(bin_file_name, "wb")
+        bin_file.write (self.generate_binary_array ())
+        bin_file.close()
+        return 0
+
+    def override_default_value(self, csv_file):
+        read_csv (self.schema, csv_file)
+        self.sync_shim_and_schema ()
+        return 0
+
+    def generate_delta_file_from_bin (self, delta_file, old_data, new_data, full=False):
+        self.load_default_from_bin (new_data, True)
+        write_csv (self.schema, delta_file, subknobs=full)
+        return 0
+
+    def generate_csv_file(self, delta_file, bin_file, bin_file2, full=False):
+        fd = open (bin_file, 'rb')
+        new_data = bytearray(fd.read())
+        fd.close()
+
+        if bin_file2 == '':
+            old_data = self.generate_binary_array()
+        else:
+            old_data = new_data
+            fd = open (bin_file2, 'rb')
+            new_data = bytearray(fd.read())
+            fd.close()
+
+        return self.generate_delta_file_from_bin (delta_file, old_data, new_data, full)
+
+
+    def load_xml (self, cfg_file):
+        self.initialize ()
+        # TODO: re-enable the xsd validation once updated
+        # dir_path = os.path.dirname(os.path.abspath(__file__))
+        # xsd = xmlschema.XMLSchema(os.path.join(dir_path, 'configschema.xsd'))
```

Comment:
I think this is where I was referring to on the xml schema validation step.

Reply:
I think if we cannot enable this step, we would not need to check in the *.xsd file, either.

Reply:
Edited the xsd file and added back the validation.